### ISSUE TITLE
pfdns should handle passthroughs case insensitively

### ISSUE THREAD

**Issue by cmammoli** [inverse-inc/packetfence#2537]
I configured an AD domain flagging "Allow on registration" and added the necessary passthroughs  but I could not join the domain from the registration vlan.

I dumped the dns traffic coming from the client and i noticed that pfdns was responding with CNAME to itself because the client was requesting "DOMAINCONTROLLER.domain" (uppercase) while I configured domaincontroller.domain (lowercase) in passthroughs.

Adding DOMAINCONTROLLER.domain in passthroughs fixed the issue.

Anyway since dns records are case insensitive I think pfdns should handle them this way.

**Comment by fdurand** [comment]
One line fix, can you try that:
```

diff --git a/lib/pf/util/dns.pm b/lib/pf/util/dns.pm
index 5ddced9..9090881 100644
--- a/lib/pf/util/dns.pm
+++ b/lib/pf/util/dns.pm
@@ -35,6 +35,7 @@ sub matches_passthrough {
     my ($domain,$zone) = @_;
     # undef domains are not passthroughs
     return ($FALSE, []) unless(defined($domain));
+    $domain = lc($domain);
 
     # Check for non-wildcard passthroughs
     my $normal = '$'.$zone.'{normal}{$domain}';

```

**Comment by cmammoli** [comment]
it works, thanks.

**Comment by jrouzierinverse** [comment]
@fdurand should this go into devel?

**Comment by fdurand** [comment]
yes

**Comment by jrouzierinverse** [comment]
Ok will push

**Comment by jrouzierinverse** [comment]
Fixed by e0713528670a937b70973824b509f7839880057e b118304c7cc3a9ee888d9423281ac99d097cc625
and pushed to maintenance
